### PR TITLE
T5 update to handle different scenario

### PR DESCRIPTION
**Pull request by sarapapi**
I propose to modify 1 argument and to add 2 arguments (optional) to the T5model class:
- model_name: becomes optional, if it is not set then T5 model without pre-trained weights is instantiated
- config: it has to be passed only if model_name is not provided (otherwise it is overwritten), it allows instantiating T5ForConditionalGeneration without passing a pre-trained model ([example](https://github.com/huggingface/transformers/issues/1714)) 
- tokenizer: optional, custom tokenizer to pass (it can be useful in case of, for example, additional tokens)

### CODE REVIEW

**Comment by ThilinaRajapakse** [comment]
I'm wondering why it would be necessary to instantiate a T5 model without pre-trained weights (the link to the HF issue is broken).

**Comment by sarapapi** [comment]
> I'm wondering why it would be necessary to instantiate a T5 model without pre-trained weights (the link to the HF issue is broken).

I think it would be useful to add the possibility to instantiate a non-pretrained model to all models (maybe for T5 it might be not so useful but for other seq2seq it might be). 
In any case, I think that passing a custom tokenizer for the training is an important feature to add for experimental purposes.

File: simpletransformers/t5/t5_model.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def chunks(lst, n):
 
 class T5Model:
     def __init__(
-        self, model_name, args=None, use_cuda=True, cuda_device=-1, **kwargs,
+        self, model_name=None, args=None, config=None, tokenizer=None, use_cuda=True, cuda_device=-1, **kwargs,
```
Comment (ThilinaRajapakse) [inline simpletransformers/t5/t5_model.py:45]: I'd be more comfortable with keeping `model_name` as a required parameter which the user can set to `None`. The idea is that you have to explicitly use a non-pretrained model if you need to.

`config` can already be passed as part of `args` so I don't think we need to add a separate parameter for it.

```suggestion
        self, model_name, args=None, tokenizer=None, use_cuda=True, cuda_device=-1, **kwargs,
```

File: simpletransformers/t5/t5_model.py [hunk 2]
```diff
@@ -91,11 +91,18 @@ def __init__(
 
         self.results = {}
 
-        self.config = T5Config.from_pretrained(model_name, **self.args.config)
-
-        self.model = T5ForConditionalGeneration.from_pretrained(model_name, config=self.config)
+        if model_name==None:
```
Comment (ThilinaRajapakse) [inline simpletransformers/t5/t5_model.py:94]:
```suggestion
        if model_name is None:
```

It will work fine without this change but this is the recommended best practice.

File: simpletransformers/t5/t5_model.py [hunk 3]
```diff
@@ -104,7 +111,10 @@ def __init__(
             self.args.fp16 = False
 
         self.args.model_type = "T5"
-        self.args.model_name = model_name
+        if model_name==None:
+            self.args.model_name = "T5fromScratch"
+        else:
+            self.args.model_name = model_name
```
Comment (ThilinaRajapakse) [inline simpletransformers/t5/t5_model.py:117]: Is this change necessary? `LanguageModelingModel` just leaves the `model_name` blank when training from scratch. I don't _think_ it will cause any issues though.

File: simpletransformers/t5/t5_model.py [hunk 4]
```diff
@@ -91,11 +91,18 @@ def __init__(
 
         self.results = {}
 
-        self.config = T5Config.from_pretrained(model_name, **self.args.config)
-
-        self.model = T5ForConditionalGeneration.from_pretrained(model_name, config=self.config)
+        if model_name==None:
+            self.config = config
+            self.model = T5ForConditionalGeneration(config=self.config)
```
Comment (ThilinaRajapakse) [inline simpletransformers/t5/t5_model.py:96]: The `config` from args can be used instead.

```suggestion
            self.config = self.args.config
            self.model = T5ForConditionalGeneration(config=self.config)
```

**Review by ThilinaRajapakse — CHANGES_REQUESTED** [review]
I think it's fine to give the option to have a model without loading pre-trained weights (both for T5 and Seq2Seq). I agree that a custom tokenizer could be useful.

I have some suggestions here. What do you think?

File: simpletransformers/t5/t5_model.py (hunk 1 above)
Comment (sarapapi) [reply simpletransformers/t5/t5_model.py:45]: Yes, I totally agree with you, maybe it can be specified in the docs that a "non-pretrained" version can be set with `None` tag.

File: simpletransformers/t5/t5_model.py (hunk 3 above)
Comment (sarapapi) [reply simpletransformers/t5/t5_model.py:117]: I've tried to leave this blank but it causes errors in saving/loading because the `model_name` is a textual part of the paths and `None` is not allowed.

Comment (ThilinaRajapakse) [reply simpletransformers/t5/t5_model.py:117]: Oh yeah, that makes sense.

Comment (ThilinaRajapakse) [inline simpletransformers/t5/t5_model.py:117]:
```suggestion
        if model_name is None:
            self.args.model_name = "T5_from_scratch"
        else:
            self.args.model_name = model_name
```

Let's make it like this then? Changed to underscores for consistency.

Comment (sarapapi) [reply simpletransformers/t5/t5_model.py]: Yes, the name was just a proposal :)

**Review by ThilinaRajapakse — APPROVED** [review]

**Comment by ThilinaRajapakse** [comment]
Awesome, thank you!
